### ISSUE TITLE
Weird support request: restoring lost data in the Nextcloud DB from the Deck Android App

### ISSUE THREAD

**Issue by shtrom** [stefan-niedermann/nextcloud-deck#1264]
# tl;dr

How can I change the local sqlite3 data to mark all cards as unsynced, so the Android app resyncs everything to the Nextcloud app?

# Context

For some reason I'm still to diagnose, my Nextcloud _server_ DB has a corrupted `oc_deck_cards` that I cannot restore (and, of course, my backups are 🤷🏼 because I ingored the issue for too long)... I have, however, a copy of the data, up-to-date, in my Deck Android app!

I am now looking at a way to resync the data from the Android app back into the Nextcloud server. What I want to do is
1. Create a new empty `oc_deck_cards` on the server side,
2. Mark all the cards in the Android app as unsynced
3. ...
4. Profit? (i.e., the Android app syncs back all the cards to the Nextcloud app).

So, question (while I'm reading the app code, and trying to answer it myself): Could this actually work? And if so, how could I do step 2? I can reasonably `adb shell` into my phone, and change the SQL DB to do what's needed to mark the cards as unsynced... but I don't know what that change should be.

**Comment by desperateCoder** [comment]
Well, that's a very special request you have here. But i think we got your back.

It should work if you remove the (remote- not local-)IDs in the db and set the sync-state to locally edited. Locally edited would cause it to sync and the missing id will treat it as a new entry and insert instead of update.

I'll update here in my lunch break in like 2.5 hours and tell you the exact steps. 

The only prerequisite is that your server app is working again properly, you can work on that in the meantime if not already done.

**Comment by desperateCoder** [comment]
Oh and a really important thing: DON'T Open the app in the meantime! If the sync is triggered and the server won't say it knows the synced data (which will be the case if your server db is blank), the app will delete the data as well!

Once the remote IDs are resetted, your data will be safe

**Comment by shtrom** [comment]
> I'll update here in my lunch break in like 2.5 hours and tell you the exact steps.

Thanks mate. Your answer is already useful. I'll have a look at the DB and see if I can make sense of it.

> DON'T Open the app in the meantime

Yup, I expected as much. At least in this case, I have a backup... Maybe I'll make a backup now...

**Comment by desperateCoder** [comment]
>  Maybe I'll make a backup now...

Excellent idea! You should be able to make a copy of your DB via ADB. In case we mess something up, we maybe can still restore everything from the last-resort copy.

I already have a plan, if the data is still in your apps DB, I'm quite convinced we can force it to work again. I'll write here later.

**Comment by desperateCoder** [comment]
OK, here you go:

``` sql
update AccessControl set id=NULL, etag=NULL, state=2;
update Activity set id=NULL, etag=NULL, state=2;
update Attachment set id=NULL, etag=NULL, state=2;
update Board set id=NULL, etag=NULL, state=2;
update Card set id=NULL, etag=NULL, state=2;
update DeckComment set id=NULL, etag=NULL, state=2;
update Label set id=NULL, etag=NULL, state=2;
update OcsProject set id=NULL, etag=NULL, state=2;
update OcsProjectResource set id=NULL, etag=NULL, state=2;
update Stack set id=NULL, etag=NULL, state=2;

update JoinBoardWithLabel set state=2;
update JoinBoardWithPermission set state=2;
update JoinBoardWithUser set state=2;
update JoinCardWithLabel set state=2;
update JoinCardWithProject set state=2;
update JoinCardWithUser set state=2;

```

that's a quite naive way i'd try first. I checked the tables, threw out anything non-related, checked for relevant fields and the above is the outcome. 

If i didn't mess up my little copy-paste session, this should trick the app to sync all up again.

There are some related tables not mentioned above and i can't gaurantee they won't mess up something, but i think it should work. It kind of depends on what the server does with it. AND if my logic is correct :sweat_smile: 

In any case: make sure to have your DB backed up, like... 5 times... :laughing: 

If anything fails or something, feel free to to post it here for further support.

**Comment by shtrom** [comment]
Cheers, I'll try this in a moment.

As I have only lost the `oc_deck_cards` table, should/could I limit myself to the `Card` and `JoinCard*` tables?
-- 
Olivier Mehani ***@***.***>
Sent from my mobile, please excuse my brevity.

**Comment by desperateCoder** [comment]
> As I have only lost the `oc_deck_cards` table,

oh, i missed that. totally correct, to get your cards back you'll need the card table and all depending tables:

``` sql
update Card set id=NULL, etag=NULL, state=2;
update DeckComment set id=NULL, etag=NULL, state=2; -- to upload all comments if you have any
update Attachment set id=NULL, etag=NULL, state=2; -- to (re-)upload and connect all attached files

update JoinCardWithLabel set state=2;
update JoinCardWithProject set state=2;
update JoinCardWithUser set state=2;
```

**Comment by shtrom** [comment]
I don't seem to have a `state` column in any of the tables. Did you mean `status`?

**Comment by desperateCoder** [comment]
> I don't seem to have a `state` column in any of the tables. Did you mean `status`?

Yep, you're right, I'm dumb. Best dev ever. 😓

**Comment by shtrom** [comment]
Well, you know what? I think that worked.

After running all those (`s/state/status`) in the sqlite DB on the phone, and 
```create table oc_deck_cards_new like oc_deck_cards;
`rename table oc_deck_cards to oc_deck_cards_bad, oc_deck_cards_new to oc_deck_cards;
```
on the server side, I restarted the app (which I had force-stopped before).

It started up and every task was marked as to sync 🔄, but that then disappeared, and 
```
select count(1) from oc_deck_cards;
+----------+
| count(1) |
+----------+
| 85       |
+----------+
1 row in set
Time: 0.367s
```

I'm spot-checking now, but this is all looking pretty good!

**Comment by desperateCoder** [comment]
That's at least some progress... You can activate the logging in the app in the settings, maybe that helps. You can post the output here if needed. But be aware that it contains your cards data, so be careful what you post.
It's nearly at the bottom of the settings.

Then you can trigger the sync and have a little bug button on the header of the sidebar. After the sync, hit the bug to retrieve the logs

**Comment by shtrom** [comment]
Nah, I think it's done. It's all there, synced back to the server 😌

**Comment by desperateCoder** [comment]
> Nah, I think it's done. It's all there, synced back to the server 😌

Aw man! Glad it worked! ❤️ 

Btw.: You nicely thought along, covering my mistakes. Wish all our users would be like that.

That was unconventional and fun! Finally the offline-capabilities I invested so much time in visibly paid off.

 If you think we're done here, feel free to close this issue and have a nice day!

**Comment by shtrom** [comment]
Yeah, to me offline caps is a must-have for this sort of apps. Otherwise there's no real point in a native app when a web view can do.

Hey, I'd like to send a token of gratitude. Should I just use the `Sponsor` link for this repo? I see the repo is not in your account. I am grateful for the original code for sure, but even more so for your help right then, so I want to make sure the love is shared. What's the best way to do this?

**Comment by desperateCoder** [comment]
Aw man, thats really not necessary, but I won't resist either 😊 

It's mostly a two-man-project, driven by me as the DB/server-communication guy and my buddy @stefan-niedermann as the UI-sorcerer. It started as two bros having a fun project and grew to... THIS.

 Its Stefan's repo and he is also managing the incomes, so @stefan-niedermann: whats the best way without paying unncessary commissions?

@shtrom your love is highly appreciated, thank you very much, mate! I also had fun solving this tho 😉

**Comment by desperateCoder** [comment]
> I'll forward all of the received money to @desperateCoder then.

Nah, gonna be shared as usual. And if it's a odd number, we'll saw that last penny in two.

**Comment by desperateCoder** [comment]
Chellange accepted. But be aware that we have Euros here, i don't have an Australian penny here to saw apart 😅 

But you'll get your photo!

**Comment by stefan-niedermann** [comment]
brot kann ja jeder brechen 🤷‍♂️

**Comment by shtrom** [comment]
🤣 cheers guys!

**Comment by shtrom** [comment]
> brot kann ja jeder brechen man_shrugging

und geld auch?

**Comment by desperateCoder** [comment]
Only if you got a money-breaking machine like i bought just for this occasion 💪 

JK, smart people would just use a metal saw and a penny coming from the other end of the world, where everything is trying to kill you 

Thanks again mate!

**Comment by cm-t** [comment]
Hi, I think i have the same need: I have the same issue, i just saw that all my export backup are empty (_only a 'format' file in the weekly nextcloud exports, because i didnt add the ` -abc` options in the scheduled sysd service, now fixed, but too late_), so only have the data (files) folder to recover :'(

i'm trying to recover 1 important deck to me. I have an Ubuntu Touch phone with [NextDeck](https://open-store.io/app/nextdeck.cloudsite), but also, in waydroid (_android container inside Ubuntu Touch_), i do have the deck f-droid app installed and synched with the datas i want.

Should i use `/home/phablet/.local/share/waydroid/data/data/it.niedermann.nextcloud.deck/databases/NC_DECK_DB.db` ?  
Is that the database file?

If yes, i could open on my desktop with a db client and update the db data. Should i do that ?:

```
update AccessControl set id=NULL, etag=NULL, state=2;
update Activity set id=NULL, etag=NULL, state=2;
update Attachment set id=NULL, etag=NULL, stastatuste=2;
update Board set id=NULL, etag=NULL, state=2;
update Card set id=NULL, etag=NULL, state=2;
update DeckComment set id=NULL, etag=NULL, state=2;
update Label set id=NULL, etag=NULL, state=2;
update OcsProject set id=NULL, etag=NULL, state=2;
update OcsProjectResource set id=NULL, etag=NULL, state=2;
update Stack set id=NULL, etag=NULL, state=2;

update JoinBoardWithLabel set state=2;
update JoinBoardWithPermission set state=2;
update JoinBoardWithUser set state=2;
update JoinCardWithLabel set state=2;
update JoinCardWithProject set state=2;
update JoinCardWithUser set state=2;

update Card set id=NULL, etag=NULL, state=2;
update DeckComment set id=NULL, etag=NULL, state=2; -- to upload all comments if you have any
update Attachment set id=NULL, etag=NULL, state=2; -- to (re-)upload and connect all attached files

update JoinCardWithLabel set state=2;
update JoinCardWithProject set state=2;
update JoinCardWithUser set state=2;

create table oc_deck_cards_new like oc_deck_cards;
rename table oc_deck_cards to oc_deck_cards_bad, oc_deck_cards_new to oc_deck_cards;
```
Should i replace all `state=` by `status=` ?  
When i've updated the tb, i supose i just replace on the pone the db files, reconnect waydroid nextcloud client app, then launch deck app and make the magic sync happend ?

The deck id was 8, i recreated a deck with the same ID on the server side, is that all i need to do?

Thank you in advance !  🙏

**Comment by desperateCoder** [comment]
Hi @cm-t ! Re-reading this issue actually made my day 😆 
> Should i use /home/phablet/.local/share/waydroid/data/data/it.niedermann.nextcloud.deck/databases/NC_DECK_DB.db ?
Is that the database file?

I don't have any experience with your setup, but the path and the filename looks exactly right. And same for you: make sure to have at least 5 copies of the original before doing anything.

> yes, i could open on my desktop with a db client and update the db data. Should i do that ?:

Yes. But the last two lines have to happen on the server side!

> Should i replace all state= by status= ?
Yes, I messed up here, as @shtrom noticed correctly.

> When i've updated the tb, i supose i just replace on the pone the db files, reconnect waydroid nextcloud client app, then launch deck app and make the magic sync happend ?

Should™ work, yes.

> The deck id was 8, i recreated a deck with the same ID on the server side, is that all i need to do?

Well, that's a different scenario then. If you do the full SQL procedure as you mentioned above, there is no need to fake the board, since from the phones point view the board is a new one. You set the remote Id to null which means "the user just created a new one". So you can have the id already in place, but you'll end up with another board with the same name, but different id.

So what's exactly broken on the server side? Is it only the card table? If yes, you can actually get away with the stripped version like @shtrom did here: https://github.com/stefan-niedermann/nextcloud-deck/issues/1264#issuecomment-1118422747

Feel free to ask further questions in case I didn't manage to make my points 😅

**Comment by cm-t** [comment]
> Hi @cm-t ! Re-reading this issue actually made my day 😆

:) glad it made you smile, and also, thank you a lot for the fast answer <3 

> I don't have any experience with your setup, but the path and the filename looks exactly right. And same for you: make sure to have at least 5 copies of the original before doing anything.

I do have many backup of that now !

> Should™ work, yes.

Almost (_not yet, but I do have hope_)


> So what's exactly broken on the server side? 

I lost the database of my Nextcloud server, and when I wanted to just import the database, I found there were no backup (_as I said, the nextcloud.export command line is well set on a systemctl timer, but without ` -abc` options it just do nothing, without any warning of that, lesson learnt, always check the backups_).

Re-installed my Nextcloud instance, added the -abc on the export systemctl service, not yet recreated all users but I do have all the files, and for cospend, I did have setup a csv monthly export that was in the files so backuped!  Now I just need to restore this deck board that I share with a team.

** **

Just tryed to reconnect Deck, but since the session was KO, it could not reconnect, and sent popup deck error.
I had to remove the account from Nextcloud client and re-add it _(all other client but this one would allow me to reconnect by web login in, i don't know really why, but that's for another time)_

Opening Deck just made a new db file.

I tryed to insert the backuped many time db file instead, but it ignored it.

So i guess I have to open both database and make a query to insert table by table :
  * User _(for this one I should make sure I re-enabled those account on the server i guess?)_
  * Board
  * UserInBoard
  * Stack
  * Label
  * Card
  * DeckComment
  * Mention
  
In that order (_to do have issue with fk_) ?
Im not used to sqlite a lot, but could i do (_using dbeaver, but can change_) something like that ?

```SQL
INSERT INTO db_new.Label 
FROM (
    SELECT *
    FROM db_bk.Label
);

INSERT INTO db_new.Card 
FROM (
    SELECT *
    FROM db_bk.Card
);
-- …
``` 

Or maybe is there an easier way ?

**Comment by desperateCoder** [comment]
I'll have to check on some details, I'll try to get this done tomorrow in my lunch break. Maybe there's a much easier way. 
Upfront: Are you sure the DB is wiped with the new login? Or is it more like you don't see any data of the account? I mean, after you connect to the new account, did you check the actual contents of the db file with DBEaver? In case you can't see the account but the data is actually there, we could just change the account id on some tables and the job is done...

**Comment by cm-t** [comment]
Thank you very much !


new db file cards (_default auto générated cards, in french_):

<img width="497" height="244" alt="Image" src="https://github.com/user-attachments/assets/b8e14707-8726-4702-8e5e-f2bb743ebfd5" />

backup db file cards:

<img width="561" height="806" alt="Image" src="https://github.com/user-attachments/assets/1c00f632-115a-4520-be67-8a1cef638122" />

I have no filter (_i just double clicked on the table, so i guess it's just select * from Cards with no where clause_), so i guess the answer is no :'(

**Comment by desperateCoder** [comment]
> for this one I should make sure I re-enabled those account on the server i guess?

Thats the first thing you should do, yes. And please make sure the users of the NEW DB look exactly like in the OLD DB, especially regarding the column `localId`. You might have to adjust the `localId` to match the OLD ones per user, this is crucial! The FKs point to the `localId` of the users, otherwise you end up with the wrong persons involved in your cards. 

I checked a few things and i'm afraid there isn't the obvious, easy way. So yes, what you mentioned above will be the way to go. So what you will have to do is the following:

1. re-login into the new account as you already did to have a fresh, NEW DB file
2. using the OLD file, update `status=2`, `id=null` and `etag=null` 
3. insert the data from the OLD backup you updated in step`2.` into the NEW DB file (i hope you had only one account on both DB files? otherwise you should check if the account id matches with the new account)
4. launch the app

regarding step `2.`:
```sql
update AccessControl set id=NULL, etag=NULL, status=2;
update Activity set id=NULL, etag=NULL, status=2;
update Attachment set id=NULL, etag=NULL, status=2;
update Board set id=NULL, etag=NULL, status=2;
update Card set id=NULL, etag=NULL, status=2;
update DeckComment set id=NULL, etag=NULL, status=2;
update Label set id=NULL, etag=NULL, status=2;
update OcsProject set id=NULL, etag=NULL, status=2;
update OcsProjectResource set id=NULL, etag=NULL, status=2;
update Stack set id=NULL, etag=NULL, status=2;

update JoinBoardWithLabel set status=2;
update JoinBoardWithPermission set status=2;
update JoinBoardWithUser set status=2;
update JoinCardWithLabel set status=2;
update JoinCardWithProject set status=2;
update JoinCardWithUser set status=2;
update JoinCardWithDependentCard set status=2;
```

regarding step `3.`:
```sql

INSERT INTO db_new.User 
FROM (
    SELECT *
    FROM db_bk.User
);

INSERT INTO db_new.Board 
FROM (
    SELECT *
    FROM db_bk.Board
);

INSERT INTO db_new.Label 
FROM (
    SELECT *
    FROM db_bk.Label
);

INSERT INTO db_new.Stack 
FROM (
    SELECT *
    FROM db_bk.Stack
);

INSERT INTO db_new.AccessControl 
FROM (
    SELECT *
    FROM db_bk.AccessControl
);

INSERT INTO db_new.OcsProject 
FROM (
    SELECT *
    FROM db_bk.OcsProject
);

INSERT INTO db_new.OcsProjectResource 
FROM (
    SELECT *
    FROM db_bk.OcsProjectResource
);


INSERT INTO db_new.Card 
FROM (
    SELECT *
    FROM db_bk.Card
);


INSERT INTO db_new.Attachment 
FROM (
    SELECT *
    FROM db_bk.Attachment
);


INSERT INTO db_new.Activity 
FROM (
    SELECT *
    FROM db_bk.Activity
);


INSERT INTO db_new.DeckComment 
FROM (
    SELECT *
    FROM db_bk.DeckComment
);

INSERT INTO db_new.JoinBoardWithLabel 
FROM (
    SELECT *
    FROM db_bk.JoinBoardWithLabel
);

INSERT INTO db_new.JoinBoardWithPermission 
FROM (
    SELECT *
    FROM db_bk.JoinBoardWithPermission
);

INSERT INTO db_new.JoinBoardWithUser 
FROM (
    SELECT *
    FROM db_bk.JoinBoardWithUser
);

INSERT INTO db_new.JoinCardWithLabel 
FROM (
    SELECT *
    FROM db_bk.JoinCardWithLabel
);

INSERT INTO db_new.JoinCardWithProject 
FROM (
    SELECT *
    FROM db_bk.JoinCardWithProject
);

INSERT INTO db_new.JoinCardWithUser 
FROM (
    SELECT *
    FROM db_bk.JoinCardWithUser
);

INSERT INTO db_new.JoinCardWithDependentCard 
FROM (
    SELECT *
    FROM db_bk.JoinCardWithDependentCard
);

```

If I didn't mess up the the order, this should run just fine. in case you got foreign key constraints, you might have to change the order of the inserts or just post it here, i hope i can instantly see whats wrong then. And i hope i didn't miss any tables, but i think it should be fine.

Please keep me updated! 😊

**Comment by cm-t** [comment]
Hi, thank again for the awnser!

this is what I've tryed:

<details>
  <summary>I've tryed this (fail)</summary>

```SQL
.open NC_DECK_DB.db AS dbbk
ATTACH 'NC_DECK_DB.3.db' AS dbnew;

INSERT INTO dbnew.User 
FROM (
    SELECT *
    FROM dbbk.User
);

INSERT INTO dbnew.Board 
FROM (
    SELECT *
    FROM dbbk.Board
);

INSERT INTO dbnew.Label 
FROM (
    SELECT *
    FROM dbbk.Label
);

INSERT INTO dbnew.Stack 
FROM (
    SELECT *
    FROM dbbk.Stack
);

INSERT INTO dbnew.AccessControl 
FROM (
    SELECT *
    FROM dbbk.AccessControl
);

INSERT INTO dbnew.OcsProject 
FROM (
    SELECT *
    FROM dbbk.OcsProject
);

INSERT INTO dbnew.OcsProjectResource 
FROM (
    SELECT *
    FROM dbbk.OcsProjectResource
);


INSERT INTO dbnew.Card 
FROM (
    SELECT *
    FROM dbbk.Card
);


INSERT INTO dbnew.Attachment 
FROM (
    SELECT *
    FROM dbbk.Attachment
);


INSERT INTO dbnew.Activity 
FROM (
    SELECT *
    FROM dbbk.Activity
);


INSERT INTO dbnew.DeckComment 
FROM (
    SELECT *
    FROM dbbk.DeckComment
);

INSERT INTO dbnew.JoinBoardWithLabel 
FROM (
    SELECT *
    FROM dbbk.JoinBoardWithLabel
);

INSERT INTO dbnew.JoinBoardWithPermission 
FROM (
    SELECT *
    FROM dbbk.JoinBoardWithPermission
);

INSERT INTO dbnew.JoinBoardWithUser 
FROM (
    SELECT *
    FROM dbbk.JoinBoardWithUser
);

INSERT INTO dbnew.JoinCardWithLabel 
FROM (
    SELECT *
    FROM dbbk.JoinCardWithLabel
);

INSERT INTO dbnew.JoinCardWithProject 
FROM (
    SELECT *
    FROM dbbk.JoinCardWithProject
);

INSERT INTO dbnew.JoinCardWithUser 
FROM (
    SELECT *
    FROM dbbk.JoinCardWithUser
);

INSERT INTO dbnew.JoinCardWithDependentCard 
FROM (
    SELECT *
    FROM dbbk.JoinCardWithDependentCard
);
```

```bash
sqlite3 <insert.sql 
extra argument: "AS"
Parse error near line 4: near "FROM": syntax error
  INSERT INTO dbnew.User  FROM (     SELECT *     FROM dbbk.User );
                          ^--- error here
Parse error near line 10: near "FROM": syntax error
  INSERT INTO dbnew.Board  FROM (     SELECT *     FROM dbbk.Board );
             error here ---^
Parse error near line 16: near "FROM": syntax error
  INSERT INTO dbnew.Label  FROM (     SELECT *     FROM dbbk.Label );
             error here ---^
Parse error near line 22: near "FROM": syntax error
  INSERT INTO dbnew.Stack  FROM (     SELECT *     FROM dbbk.Stack );
             error here ---^
Parse error near line 28: near "FROM": syntax error
  INSERT INTO dbnew.AccessControl  FROM (     SELECT *     FROM dbbk.AccessContr
                     error here ---^
Parse error near line 34: near "FROM": syntax error
  INSERT INTO dbnew.OcsProject  FROM (     SELECT *     FROM dbbk.OcsProject );
                  error here ---^
Parse error near line 40: near "FROM": syntax error
  INSERT INTO dbnew.OcsProjectResource  FROM (     SELECT *     FROM dbbk.OcsPro
                          error here ---^
Parse error near line 47: near "FROM": syntax error
  INSERT INTO dbnew.Card  FROM (     SELECT *     FROM dbbk.Card );
                          ^--- error here
Parse error near line 54: near "FROM": syntax error
  INSERT INTO dbnew.Attachment  FROM (     SELECT *     FROM dbbk.Attachment );
                  error here ---^
Parse error near line 61: near "FROM": syntax error
  INSERT INTO dbnew.Activity  FROM (     SELECT *     FROM dbbk.Activity );
                error here ---^
Parse error near line 68: near "FROM": syntax error
  INSERT INTO dbnew.DeckComment  FROM (     SELECT *     FROM dbbk.DeckComment )
                   error here ---^
Parse error near line 74: near "FROM": syntax error
  INSERT INTO dbnew.JoinBoardWithLabel  FROM (     SELECT *     FROM dbbk.JoinBo
                          error here ---^
Parse error near line 80: near "FROM": syntax error
  INSERT INTO dbnew.JoinBoardWithPermission  FROM (     SELECT *     FROM dbbk.J
                               error here ---^
Parse error near line 86: near "FROM": syntax error
  INSERT INTO dbnew.JoinBoardWithUser  FROM (     SELECT *     FROM dbbk.JoinBoa
                         error here ---^
Parse error near line 92: near "FROM": syntax error
  INSERT INTO dbnew.JoinCardWithLabel  FROM (     SELECT *     FROM dbbk.JoinCar
                         error here ---^
Parse error near line 98: near "FROM": syntax error
  INSERT INTO dbnew.JoinCardWithProject  FROM (     SELECT *     FROM dbbk.JoinC
                           error here ---^
Parse error near line 104: near "FROM": syntax error
  INSERT INTO dbnew.JoinCardWithUser  FROM (     SELECT *     FROM dbbk.JoinCard
                        error here ---^
Parse error near line 110: near "FROM": syntax error
  INSERT INTO dbnew.JoinCardWithDependentCard  FROM (     SELECT *     FROM dbbk
                                 error here ---^
```
</details>

and 
<details>
  <summary>I also tryied this (fail)</summary>
  


```bash
sudo apt install sqlite3
 sqlite3 NC_DECK_DB.db ".dump User" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump Board" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump Label" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump Stack" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump AccessControl" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump Stack" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump AccessControl" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump OcsProject" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump OcsProjectResource" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump Card" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump Attachment" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump Activity" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump DeckComment" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump JoinBoardWithLabel" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump JoinBoardWithPermission" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump JoinBoardWithUser" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump JoinCardWithLabel" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump JoinCardWithProject" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump JoinCardWithUser" | sqlite3 NC_DECK_DB.2.db 
 sqlite3 NC_DECK_DB.db ".dump JoinCardWithDependentCard" | sqlite3 NC_DECK_DB.2.db 
```

```bash
Parse error near line 3: table `Board` already exists
  CREATE TABLE `Board` (`title` TEXT, `ownerId` INTEGER NOT NULL, `color` INTEGE
               ^--- error here
Runtime error near line 4: UNIQUE constraint failed: Board.localId (19)
Runtime error near line 5: UNIQUE constraint failed: Board.localId (19)
Runtime error near line 6: UNIQUE constraint failed: Board.localId (19)
Runtime error near line 7: UNIQUE constraint failed: Board.localId (19)
Runtime error near line 8: UNIQUE constraint failed: Board.localId (19)
(…)
```
</details>

I have error too (because it tryes to create table without a  "if not exist" condition, but then it seems to do things after, but, constraints fails, so maybe i should check the data closer.

I post this here so i will remember easyer where to start from,  in 2 weeks (_I will be offline for holidays_)

to be continued :)

**Comment by desperateCoder** [comment]
@cm-t jfyi, we're about to make a feature out of this. In the meantime I got another lost soul having this kind of issue, so i added a feature in the settings, so you can do that with a few taps: 

<img width="403" height="896" alt="Image" src="https://github.com/user-attachments/assets/3cae0306-1f14-43f7-8eaf-72c898272fda" />

<img width="396" height="906" alt="Image" src="https://github.com/user-attachments/assets/5310e9d8-1e96-47fe-a2c8-4747c28c8d86" />

**Comment by cm-t** [comment]
Wow that's great ! Thank you <3

This update would definitely resolve this use case
